HIRS_Utils/src/main/java/hirs/data/persist/certificate/Certificate.java

```diff
@@ -328,6 +331,47 @@ public Certificate(final byte[] certificateBytes) throws IOException {
         this.certAndTypeHash = Objects.hash(certificateHash, getClass().getSimpleName());
     }
 
+    @SuppressWarnings("magicnumber")
+    private byte[] trimCertificate(final byte[] certificateBytes) {
+        int certificateStart = 0;
+        int certificateLength = 0;
+
+        StringBuilder malformedCertStringBuilder = new StringBuilder(MALFORMED_CERT_MESSAGE);
+        for (int i = 0; i < certificateBytes.length; i++) {
+            if (i == certificateBytes.length - 1) {
+                throw new IllegalArgumentException(malformedCertStringBuilder
+                        .append(" No certificate length field could be found.").toString());
+            }
+
+            // Look for first ASN.1 Sequence marked by the two bytes (0x30) and (0x82)
+            if (certificateBytes[i] == 0x30 && certificateBytes[i + 1] == (byte) 0x82) {
+                if (i >= certificateBytes.length - 3) {
+                    throw new IllegalArgumentException(malformedCertStringBuilder
+                            .append(" Certificate is nothing more than ASN.1 Sequence.")
+                            .toString());
+                }
+                // Mark the start of the first ASN.1 Sequence / Certificate Body
+                certificateStart = i;
+
+                // Parse the length as the 2-bytes following the start of the ASN.1 Sequence
+                certificateLength = (certificateBytes[i + 2] & 0xFF) << 8;
```

Comment:
Since Java doesn't have a concept of an unsigned value, we have to coerce the byte value from certificateBytes[i + 2] up to integer to prevent a byte overflow occurring and ending up with a large negative value for the certificateLength.